app/src/test/groovy/pl/targosz/invoicing/database/RepositoryTest.groovy

```diff
@@ -0,0 +1,127 @@
+package pl.targosz.invoicing.database
+
+import pl.targosz.invoicing.model.Company
+import pl.targosz.invoicing.model.Invoice
+import pl.targosz.invoicing.model.InvoiceEntry
+import pl.targosz.invoicing.model.Vat
+import spock.lang.Specification
+import java.time.LocalDateTime;
+
+
+abstract class RepositoryTest extends Specification {
+
+    InvoiceRepository repository
+
+    abstract InvoiceRepository getInvoiceRepository()
+
+    def createdAt = LocalDateTime.now()
+    def firstEntry = new InvoiceEntry("Dom", 1000000 as BigDecimal, 23 as BigDecimal, Vat.VAT_23)
+    def buyer = new Company("1234567891", "KashwayCompany", "Ul. XYZ 15/7 Warszawa")
+    def buyer2 = new Company("456789123", "KFC", "Ul. Fiołkowa 15, Kraków")
+    def seller = new Company("3373187493", "Mateusz Idiks", "Al. XYZ 5, POZNAN")
+    def seller2 = new Company("7515212312", "Hubert Targosz", "Al. 3 Wieszczy 17/5, Kraków")
+    def entries1 = Arrays.asList(firstEntry)
+    UUID id
```

Comment:
I don't think this global variable is necessary

Reply:
Done